src/main/java/com/laytonsmith/core/natives/interfaces/Mixed.java

```diff
@@ -104,66 +109,59 @@
 
 	/**
 	 * Generally speaking, we cannot use Java's instanceof keyword to determine if something is an instanceof, because
-	 * user classes do not extend the hierarchy of objects in MethodScript. Essentially, we need to extend Java's
+	 * user classes do not extend the hierarchy of objects in MethodScript.Essentially, we need to extend Java's
 	 * instanceof keyword, so in order to do that, we must compare objects with a custom method, rather than rely on
 	 * Java's keyword.
 	 *
-	 * This method works with CClassTypes.
-	 *
-	 * Implementation note: The implementation of this should just be
-	 * {@link Construct#isInstanceof(com.laytonsmith.core.natives.interfaces.Mixed,
-	 * com.laytonsmith.core.constructs.CClassType)} which supports Mixed values.
-	 *
-	 * @param type
+	 * @param type The CClassType to check against
+	 * @param lhsGenericParameters The LHS generic definition (may be null)
+	 * @param env The environment
 	 * @return
 	 */
-	public boolean isInstanceOf(CClassType type);
+	public boolean isInstanceOf(CClassType type, LeftHandGenericUse lhsGenericParameters, Environment env);
 
 	/**
-	 * Generally speaking, we cannot use Java's instanceof keyword to determine if something is an instanceof, because
-	 * user classes do not extend the hierarchy of objects in MethodScript. Essentially, we need to extend Java's
-	 * instanceof keyword, so in order to do that, we must compare objects with a custom method, rather than rely on
-	 * Java's keyword.
-	 *
-	 * This method works with class type directly.
-	 *
-	 * Implementation note: The implementation of this should just be
-	 * {@link Construct#isInstanceof(com.laytonsmith.core.natives.interfaces.Mixed, java.lang.Class)} which supports
-	 * Mixed values.
-	 *
-	 * @param type
-	 * @return
-	 */
-	public boolean isInstanceOf(Class<? extends Mixed> type);
```

Comment:
Breaks a few extensions, though some still use plain old instanceof.

Reply:
As discussed, extensions can use InstanceofUtil.isInstanceof (or just java instanceof), which exists today, and will continue to exist. This will need to be updated now, but can be done immediately.